.github/ISSUE_TEMPLATE/design_issue.md

```diff
@@ -0,0 +1,29 @@
+---
+name: Design Issue
+about: Code has an architecture or design issue
+title: ''
+labels: design
+assignees: ''
+
+---
+## Summary
+
+<!-- Describe the issue in a few sentences. -->
+
+## Current Design
+
+<!-- Describe how the current version of relevant code works. -->
```

Comment:
Maybe adding some reference to the code would be useful.

Reply:
```suggestion
<!-- Describe how the current version of relevant code works.
     Refer to specific files/lines or specific packages/methods where
     applicable. -->
```
Like this?